### PR TITLE
Allow setting visibility on cargo_build_scripts

### PR DESCRIPTION
**Pull request by illicitonion**

### CODE REVIEW

File: cargo/cargo_build_script.bzl [hunk 1]
```diff
@@ -244,6 +244,7 @@ def cargo_build_script(
         data = [],
         links = None,
         rustc_env = {},
+        visibility = None,
```
Comment (UebelAndre) [inline cargo/cargo_build_script.bzl:247]: Why not pull `visibility` out of `kwargs`? 

Also, what about `tags` if you're already in this neck of the woods 😄 I'm never sure how that should best be handled in macros which generate multiple targets. Though, it's fine if that isn't included in this PR

**Review by UebelAndre — CHANGES_REQUESTED** [review]
Good idea!

One question, and buildifier/docs seems unhappy

File: cargo/cargo_build_script.bzl (hunk 1 above)
Comment (illicitonion) [reply cargo/cargo_build_script.bzl:247]: I don't have valid use-cases in mind for anyone wanting to set visibility for the internal generated target whose only purpose is to run the build script, so I don't see why we should let people set that visibility unless someone _does_ have a use-case for it :)

Added tags under the same idea - we can introduce a separate `binary_tags` or similar if we need to in the future.

Comment (UebelAndre) [reply cargo/cargo_build_script.bzl:247]: I think at the very least, the underlying target should be tagged as `manual` then. It seems like it should only ever be build if it's used in the runner target. Is that something you'd be willing to add to this PR?

Comment (illicitonion) [reply cargo/cargo_build_script.bzl:247]: Done

Comment (UebelAndre) [reply cargo/cargo_build_script.bzl:247]: Sorry, what was done? Feels like a commit is missing, maybe?

Comment (illicitonion) [reply cargo/cargo_build_script.bzl:247]: Yeah, there's a whole "maybe add a manual tag to tags" thing?

Comment (illicitonion) [reply cargo/cargo_build_script.bzl:247]: Oh which I squashed into the regenerate documentation commit. Ugh.

Comment (UebelAndre) [reply cargo/cargo_build_script.bzl:247]: I still don't see that change. Looking at the latest commit on your [cbs-visibility](https://github.com/illicitonion/rules_rust/blob/c982010184d94a6504ee434718ff67bcdcf0c723/cargo/cargo_build_script.bzl#L330-L338) branch...

Comment (illicitonion) [reply cargo/cargo_build_script.bzl:247]: Oops, weirdness happened :) Should look better now!

**Review by UebelAndre — APPROVED** [review]
Awesome! Thanks!!

**Pull request merged**